samza-core/src/main/scala/org/apache/samza/diagnostics/DiagnosticsStreamMessage.java

```diff
@@ -155,6 +158,14 @@ public void addProcessorStopEvents(List<ProcessorStopEvent> stopEventList) {
     }
   }
 
+  /**
+   * Add the job's config to the message.
+   * @param config the config to add.
+   */
+  public void addConfig(Config config) {
+    addToMetricsMessage(GROUP_NAME_FOR_DIAGNOSTICS_MANAGER, CONFIG, config);
```

Comment:
config is a non-primitive object, cast to Map<String, String> here before serializing and adding into the map; similar to how containerModels are added above.